README.md
# blog2019-04-26
blog2019-04-26 is a blog application made with Laravel 5.7, and Vue 2 . You can follow the steps in [INSTALL.md](INSTALL.md) , and test it in your local computer. Not suitable for production use.


### Installation Instructions

See [INSTALL.md](INSTALL.md) for instructions.


## What can I do with this blog application ?
You can create blog posts with this application. 
  - Supports markdown format.
  - You can create, edit, view, delete, list blog posts.
  - TODO: Can not add photos at his time. In progress.
  - TODO: Can not add videos at this time. In progress.
  - You can enter information about yourself.
    - Can add a photo.
    - You can enter paragraphs about yourself.
    - Emails.
    - Social media links.
  - Login options
    - Form based login
    - Login with Social Media accounts (OAuth)
      - Google, Github ...

## Usage

### [Create a user](docs/create_user.md)

### [Add user to a role](docs/add_role.md)

### [Insert sample data](docs/insert_sample.md)

## Screenshots

- [Mainpage](docs/images/mainpage.md)
- [Posts](docs/images/posts.md)
  - [Read Post](docs/images/post_view.md)
- [About](docs/images/about.md)
- Manage
    - [Posts](docs/images/manage_posts.md)
      - [New](docs/images/manage_posts_new.md), [View](docs/images/manage_posts_view.md), [Update](docs/images/manage_posts_update.md), [Delete](docs/images/manage_posts_delete.md)
  - [Profile](docs/images/manage_profile.md)
  - [Social Networks](docs/images/manage_social_networks.md)
  - Login
    - [Form based](docs/images/form_login.md)
    - [Google Signin](docs/images/google_signin.md)
    

## License

Copyright 2023 Recep GÜÇLÜER <rgucluer@gmail.com> .
blog2019-04-26 is [MIT licensed](LICENSE) .

## Licenses

[licenses.md](licenses.md) contains the name, version and commit sha, description, homepage, and license information for every php dependency in this project.

## About sample texts & photos in screenshots
Sample text is created with ChatGPT: [https://chat.openai.com/](https://chat.openai.com/)

Any text, photo used in these screenshots are for demonstration purposes only. There is no relation to people or companies in the real world.

Stock photos downloaded, and used with little modifications from [pexels.com](pexels.com) . Thanks to pexels.com, and all the creators sharing their content with the community.

| Photo | Creator | From Creator |
| ----- | ------- | ------------ |
| [pexels-anna-shvets-3727508.jpg](https://www.pexels.com/photo/woman-in-white-dress-shirt-wearing-eyeglasses-3727508/) | [Anna Shvets](https://www.pexels.com/@shvetsa/) | Your donations inspire me to create |
| [pexels-mikhail-nilov-8871884.jpg](https://www.pexels.com/photo/woman-in-black-dress-shirt-hand-on-her-pocket-8871884/) | [Mikhail Nilov](https://www.pexels.com/@mikhail-nilov/) | DONATE PLEASE ! Your donations inspire us to create!  |
| [pexels-vazhnik-7562313.jpg](https://www.pexels.com/photo/man-wearing-eyeglasses-on-white-background-7562313/) | [VAZHNIK](https://www.pexels.com/@vazhnik/) | Thanks for your donations! |
| [pexels-chris-f-7989569.jpg](https://www.pexels.com/photo/man-in-green-cap-sitting-near-concrete-wall-7989569/) | [Chris F](https://www.pexels.com/@chris-f-38966/) | Designer & Photographer for Magnum Press |

These texts, and photos are not in the source code, or in the resources of this project. They are used in the sample blog application only. Results are the screenhots.


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

## Current Abilities
- You can create blog posts with this application
  - Supports markdown format
  - You can create, edit, view, delete, list blog posts
  - TODO: Can not add photos at his time. In progress
  - TODO: Can not add videos at this time. In progress
  - You can enter information about yourself
    - Can add a photo
    - You can enter paragraphs about yourself
    - Emails
    - Social media links
  - Login options
    - Form based login
    - Login with Social Media accounts (OAuth)
      - Google, Github ...
---

## [0.6.1]

### Added
- Add CHANGELOG.md

---

## [0.6.0] - 2023-08-17

### Changed
- Set license to MIT license #([74a297c](https://github.com/oldblogs/blog2019-04-26/commit/74a297c68d36a278261951e9601c7eee856fc0ce))  2023-08-17
- Tidy previous commit messages. Commit dates affected because of git rebase #([95f6a3f](https://github.com/oldblogs/blog2019-04-26/commit/95f6a3fbb99726d199a32ed45a4ad1890eb25fda))  2023-08-17
  - Merge branch 'migrate_semver' 

---

## [0.5.23] - 2023-08-08

### Changed
- More info on composer.json. Add licenses.md #([959a9e9](https://github.com/oldblogs/blog2019-04-26/commit/959a9e9ff619189549ccba0a4bad89dba17432da))  2023-08-10
  - Add more information about user, and project
  - licenses.md
- composer update #([22fe452](https://github.com/oldblogs/blog2019-04-26/commit/22fe452f5ddcd000c28c045e06cba254b4241b6f))  2023-08-08
- Change composer project name to oldblogs/blog2019-04-26 #([b8849ea](https://github.com/oldblogs/blog2019-04-26/commit/b8849eae294568e2211106ac9dabb922fd09a11c))  2023-08-08

---

## [0.5.22] - 2023-08-08
### Added
- Files added #([af9615c](https://github.com/oldblogs/blog2019-04-26/commit/af9615cf5f6f9cd7d12c0bd529e4c0dd2b403409))  2023-08-08 00:37
  - docs/images/manage_posts_view.jpg
  - docs/images/manage_posts_view.md
  - docs/images/post_view.jpg
  - docs/images/post_view.md
  - profile.png

### Changed
- Add NOTICE file, edit README.md #([(95ec63c)](https://github.com/oldblogs/blog2019-04-26/commit/95ec63c6c7b937dc33c61fe5937bb08216b2b9b9))  2023-08-08 10:07

- Remove extension of license file #([61111d9](https://github.com/oldblogs/blog2019-04-26/commit/61111d9355e84ee8bcb1acf63afbaaa5df37eb3e))  2023-08-08 09:22
  - rename: LICENSE.md -> LICENSE

- Improve documentation #([5b962c2](https://github.com/oldblogs/blog2019-04-26/commit/5b962c23687f1aee5dba949e8a037e63b6deb718))  2023-08-08 01:56
  - docs/images/about.jpg

- Improve documentation #([af9615c](https://github.com/oldblogs/blog2019-04-26/commit/af9615cf5f6f9cd7d12c0bd529e4c0dd2b403409))  2023-08-08 00:37
  - README.md

- Improve documentation #([6af7362](https://github.com/oldblogs/blog2019-04-26/commit/6af7362a123cdf511b6cf8cd41e2ee730996f6b1))  2023-08-07 17:49
  - About sample texts & photos in screenshots
  - resources/views/frontview/page/posts/show.blade.php

- Minor change in documentation: README.md #([b51391c](https://github.com/oldblogs/blog2019-04-26/commit/b51391c43e82ef85b2db9937c1083f21ef8921ce))  2023-08-07 15:35

### Removed
- Delete file #([bbb55f4](https://github.com/oldblogs/blog2019-04-26/commit/bbb55f486e8182c3d6d0fc17771f9824cb2b537a))  2023-08-08 00:38

---

## [0.5.21] - 2023-08-07

### Changed
- About Section. Improve documentation #([dafbefc](https://github.com/oldblogs/blog2019-04-26/commit/dafbefc0796c41162c819d925c359c7a3771e582)) 2023-08-07 15:14
  - docs/images/about.jpg
  - docs/images/mainpage.jpg
  - docs/images/manage_profile.jpg
  - docs/images/posts.jpg
- Use user's name in About Section.
  - resources/views/frontview/page/posts/show.blade.php
  - resources/views/frontview/partial/postauthor.blade.php
  - resources/views/layouts/app.blade.php

---

## [0.5.20] - 2023-08-06

### Added
- Create LICENSE.md  #([b39a571](https://github.com/oldblogs/blog2019-04-26/commit/b39a5717dada59462e9405dbdf3aaac92cac3b8a))  2023-08-06 19:15

### Changed
- Add copyright and license information to README.md #([1584c9d](https://github.com/oldblogs/blog2019-04-26/commit/1584c9d8610692f37a95fd77430e830327e636d7))  2023-08-06 19:20

---

## [0.5.19] - 2023-08-06

### Added
- Improve documentation #([b1b9517](https://github.com/oldblogs/blog2019-04-26/commit/b1b9517e061030e0679cd0803526ca5ac3f2e092))  2023-08-06 16:02
  - docs/images/form_login.jpg
  - docs/images/form_login.md
  - docs/images/google_signin.jpg
  - docs/images/google_signin.md
  - docs/images/manage_posts.jpg
  - docs/images/manage_posts.md
  - docs/images/manage_posts_delete.jpg
  - docs/images/manage_posts_delete.md
  - docs/images/manage_posts_new.jpg
  - docs/images/manage_posts_new.md
  - docs/images/manage_posts_update.jpg
  - docs/images/manage_posts_update.md
  - docs/images/manage_profile.jpg
  - docs/images/manage_profile.md
  - docs/images/manage_social_networks.jpg
  - docs/images/manage_social_networks.md

### Changed
- Improve documentation
  - README.md

---

## [0.5.18] - 2023-08-05

#([9c5303f](https://github.com/oldblogs/blog2019-04-26/commit/9c5303f06447b2cf31357358ec2c9adb40c5f147))  2023-08-05

### Added
- Improve documentation
  - docs/images/about.jpg
  - docs/images/about.md
  - docs/images/mainpage.md
  - docs/images/posts.jpg

### Changed
- Improve documentation
  - README.md
  - docs/add_role.md
  - docs/artisan_cache.md
  - docs/create_user.md
  - docs/insert_sample.md

---

## [0.5.17] - 2023-08-05

### Changed
- Improve documentation #([9a059ec](https://github.com/oldblogs/blog2019-04-26/commit/9a059ece476f2be407b44f50835f59888f59778d))  2023-08-05
  - INSTALL.md
  - docs/add_role.md
  - docs/apache.md
  - docs/composer.md
  - docs/create_user.md
  - docs/file_permissions.md
  - docs/images/registration_01_send_invitation.md
  - docs/images/registration_02_read_email.md
  - docs/images/registration_03_register.md
  - docs/images/registration_04_check_mail.md

routes/web.php
<?php

Route::get('register/{code}/{email}', 'Auth\RegisterController@showRegistrationForm')->name('getregister');
Route::post('register/{code}/{email}', 'Auth\RegisterController@register')->name('postregister');

// E-mail verification routes
Route::get('email/verify',      'Auth\VerificationController@show')->name('verification.notice');
Route::get('email/verify/{id}', 'Auth\VerificationController@verify')->name('verification.verify');
Route::get('email/resend',      'Auth\VerificationController@resend')->name('verification.resend');     

// Authentication Routes...
if ( config('app.form_login') ) {
  Route::get('login', 'Auth\LoginController@showLoginForm')->name('login');
  Route::post('login', 'Auth\LoginController@login');
  
  // Blog admin makes password reset request with terminal artisan command :
  // php artisan user:reset-password <user email>
  // An email is sent to the user with a link.
    
  // Password Reset Routes...
  Route::get('password/reset/{token}', 'Auth\ResetPasswordController@showResetForm')->name('password.reset');
  Route::post('password/reset', 'Auth\ResetPasswordController@reset')->name('password.reset.process');

} else {
  if( 'form_login' === config('app.default_auth') ){
    // When we disable form_login , we must also change the default_auth in .env
    // TODO: Log error
    abort(500, 'Error: Misconfiguration.');
  } else {
    // default_auth ' s first letter is made uppercase and route is formed
    // TODO: check for strings starting with letter i
    $routetg = 'Auth\\'                . ucfirst( config('app.default_auth') ) .
               'Controller@redirectTo' . ucfirst( config('app.default_auth') );
    Route::get('login', $routetg)->name('login');
  }
}

// ------------------------------------------------------------
// Social Logins
// ------------------------------------------------------------
// https://laravel.com/docs/5.7/socialite


// Google Sign In
// https://hdtuto.com/article/laravel-56-login-with-google-with-socialite
// TODO : Check routes exist
if( null !== config('services.google.client_id')  ) {

  Route::get( config('services.google.auth_endpoint') ,
    'Auth\GoogleController@redirectToGoogle')->name('google_auth_endpoint');

  Route::get( config('services.google.token_exchange_endpoint') ,
    'Auth\GoogleController@handleGoogleCallback');
}

// Github Sign In
//
// TODO : Check routes exist
if( null !== config('services.github.client_id')  ) {

  Route::get( config('services.github.auth_endpoint') ,
    'Auth\GithubController@redirectToGithub')->name('github_auth_endpoint');

  Route::get( config('services.github.token_exchange_endpoint') ,
    'Auth\GithubController@handleGithubCallback');
}

Route::post('logout', 'Auth\LoginController@logout')->name('logout');


// ------------------------------------------------------------
// Front view - Pages that non administrator visitors see
// ------------------------------------------------------------

// Main page
Route::get('/', 'PostController@mainpage')->name('mainpage');

// List Posts
Route::get('/posts', 'PostController@index');

// Get a single Post
Route::get('/posts/{post}', 'PostController@show');

// About page
Route::get('/about', 'AboutController@index')->name('about');

// ------------------------------------------------------------
// Manage View - Pages that registered users see.
//   Every registered users can see /profile page.
//   Authorization needed for certain pages.
// ------------------------------------------------------------

$manage_route = '/'.config('app.management', 'manage');

// User Profile Page Web
Route::middleware('auth:web', 'verified')->get('/profile',
  'ProfileController@index')->name('profile');

// Blog Management Page
Route::middleware('auth:web', 'role:admin')
  ->get($manage_route, 'DashboardController@index')
  ->name('dashboard');

// ------------------------------------------------------------
// Csocials
// ------------------------------------------------------------

// Get Csocials list in manage view
Route::middleware('auth:web', 'role:admin')
  ->get($manage_route.'/csocials',  'CsocialController@index')
  ->name('csocials.index');

// Get New Csocial Form
Route::middleware('auth:web', 'role:admin')
  ->get($manage_route.'/csocials/create', 'CsocialController@create')
  ->name('csocials.create');

// Create a new contact record with given form data
Route::middleware('auth:web', 'role:admin')
  ->post($manage_route.'/csocials', 'CsocialController@store')
  ->name('csocials.store');

// Show an contact record in the manage view
Route::middleware('auth:web', 'role:admin')
  ->get($manage_route.'/csocials/{csocial}',  'CsocialController@show')
  ->name('csocials.show');

// Edit an contact record
Route::middleware('auth:web', 'role:admin')
  ->get($manage_route.'/csocials/{csocial}/edit', 'CsocialController@edit')
  ->name('csocials.edit');

// Update an contact record
Route::middleware('auth:web', 'role:admin')
  ->patch($manage_route.'/csocials/{csocial}',  'CsocialController@update')
  ->name('csocials.update');

// Delete an contact record
Route::middleware('auth:web', 'role:admin')
  ->delete($manage_route.'/csocials/{csocial}', 'CsocialController@delete')
  ->name('csocials.delete');

